Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp

```diff
@@ -279,27 +286,27 @@ void GStreamerRegistryScanner::fillMimeTypeSetFromCapsMapping(const GStreamerReg
 
 void GStreamerRegistryScanner::initializeDecoders(const GStreamerRegistryScanner::ElementFactories& factories)
 {
-    if (factories.hasElementForMediaType(ElementFactories::Type::AudioDecoder, "audio/mpeg, mpegversion=(int)4")) {
+    if (auto result = factories.hasElementForMediaType(ElementFactories::Type::AudioDecoder, "audio/mpeg, mpegversion=(int)4")) {
         m_decoderMimeTypeSet.add(AtomString("audio/aac"_s));
         m_decoderMimeTypeSet.add(AtomString("audio/mp4"_s));
         m_decoderMimeTypeSet.add(AtomString("audio/x-m4a"_s));
         m_decoderMimeTypeSet.add(AtomString("audio/mpeg"_s));
         m_decoderMimeTypeSet.add(AtomString("audio/x-mpeg"_s));
-        m_decoderCodecMap.add(AtomString("mpeg"_s), false);
-        m_decoderCodecMap.add(AtomString("mp4a*"_s), false);
+        m_decoderCodecMap.add(AtomString("mpeg"_s), result);
+        m_decoderCodecMap.add(AtomString("mp4a*"_s), result);
```

Comment:
That `false` meant "not using hardware acceleration". `result` having its bool operator return true means we have an element matching the media type. We now additionally store the result in the codec map, that's all, I don't think this is a change of behavior.

Reply:
What I mean is that you're passing a result that is true, otherwise you would not have entered that if branch, as result is checked and has to be true to enter that if branch, right?

The case of other branches below is a bit different because you passed before the hardware support, but now you also pass a variable that you check in the if branch, being always true.

Reply:
The result evaluates to `true` but is a struct...